### ISSUE TITLE
Crash when trying to run some of the examples

### ISSUE THREAD

**Issue by pabogdan** [nengo/nengo_gui#603]
From what I can tell, all of the examples built-into nengo_gui involving SPA crash the server with the following error (other examples do too):
```bash
Traceback (most recent call last):
  File "/home/bogdanp/nengo_gui/nengo_gui/swi.py", line 276, in handle_request
    text = getattr(self, command)(*args[1:], **db)
  File "/home/bogdanp/nengo_gui/nengo_gui/server.py", line 79, in swi
    page = self.server.gui.create_page(filename, reset_cfg=reset_cfg)
  File "/home/bogdanp/nengo_gui/nengo_gui/gui.py", line 103, in create_page
    page = nengo_gui.page.Page(self, filename=filename, reset_cfg=reset_cfg)
  File "/home/bogdanp/nengo_gui/nengo_gui/page.py", line 95, in __init__
    self.load()
  File "/home/bogdanp/nengo_gui/nengo_gui/page.py", line 153, in load
    self.name_finder = nengo_gui.NameFinder(self.locals, self.model)
  File "/home/bogdanp/nengo_gui/nengo_gui/namefinder.py", line 13, in __init__
    self.find_names(net)
  File "/home/bogdanp/nengo_gui/nengo_gui/namefinder.py", line 36, in find_names
    self.known_name[attr] = '%s.%s' % (net_name, class_attr)
NameError: global name 'class_attr' is not defined
```

**Comment by pabogdan** [comment]
A bit of trimming of one of the examples suggests that this is an issue when trying to add an element to the SPA network:
```python
model = spa.SPA()
with model:
    cortex = spa.Buffer(dimensions=dimensions)
```
yields the following error
```bash
Traceback (most recent call last):
  File "/home/bogdanp/nengo_gui/nengo_gui/page.py", line 203, in execute
    exec(code, locals)
  File "<string>", line 10, in <module>
  File "/usr/local/lib/python2.7/dist-packages/nengo-2.1.0_dev-py2.7.egg/nengo/spa/spa.py", line 130, in __exit__
    % (net))
ValueError: <Buffer (unlabeled) at 0x7fdad8854a10> was not added as an attribute of the SPA network and won't be detected
```
but trying to add it to the model:
```python
model = spa.SPA()
with model:
    model.cortex = spa.Buffer(dimensions=dimensions)
```
causes the error described initially.

**Comment by Seanny123** [comment]
Fixed it in #604
